configs.py

```diff
@@ -21,6 +22,25 @@ def get_mean_std(dataset):
 
 def get_datasets(dataset, augment_clf_train=False, add_indices_to_data=False, num_positive=None,
                  augment_test=False, train_proportion=1.):
+    if dataset == 'spirograph':
+        return get_spirograph_dataset()
+    else:
+        return get_img_datasets(dataset, augment_clf_train, add_indices_to_data, num_positive,
+                                augment_test, train_proportion)
+
+
+def get_spirograph_dataset(augment_clf_train=False, add_indices_to_data=False, num_positive=None,
+                           augment_test=False, train_proportion=1.):
+
+    spirograph = DrawSpirograph(['m', 'b', 'sigma', 'rfore'], ['h', 'rback', 'gfore', 'gback', 'bfore', 'bback'])
```

Comment:
Here, we define which parameters are treated as nuisance (aka augmentation) parameters, and which are considered generative factors.

Reply:
I could be interesting to include `b` or `h` within the augmentations